pkg/agent/inteface.go

```diff
@@ -0,0 +1,53 @@
+package agent
+
+import (
+	"crypto/x509/pkix"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// AgentAddon defines manifests of agent deployed on managed cluster
+type AgentAddon interface {
+	// Manifests returns a list of manifest resources to be deployed on the managed cluster for this addon
+	Manifests(cluster *clusterv1.ManagedCluster, config runtime.Object) ([]runtime.Object, error)
+
+	// GetAgentAddonOptions returns the agent options.
+	GetAgentAddonOptions() AgentAddonOptions
+}
+
+// AgentAddonWithRegistration manifests of agent deployed on managed cluster, rbac rules and bootstrap secret for agent to register to hub.
+type AgentAddonWithRegistration interface {
+	AgentAddon
+	// HubRBAC returns the role and rolebinding on the hub for an agent on the managed cluster.
+	HubRBAC(cluster *clusterv1.ManagedCluster, group string) (*rbacv1.Role, *rbacv1.RoleBinding)
```

Comment:
Do we need to support clusterrole/clusterrolebindnig?

Reply:
added